ebean-core/src/test/java/org/tests/query/TestImplicitJoinOnParentRelationship.java

```diff
@@ -24,11 +24,11 @@ public void test() {
     query.findList();
 
     if (isPostgres()) {
-      String expectedSql = "select distinct on (t0.id) t0.id, t0.name from o_customer t0 join o_order u1 on u1.kcustomer_id = t0.id join o_order_detail u2 on u2.order_id = u1.id join o_product u3 on u3.id = u2.product_id where u3.name = ?";
+      String expectedSql = "select distinct on (t0.id) t0.id, t0.name from o_customer t0 join o_order u1 on u1.kcustomer_id = t0.id and u1.order_date is not null join o_order_detail u2 on u2.order_id = u1.id and u2.id > 0 join o_product u3 on u3.id = u2.product_id where u3.name = ?";
```

Comment:
The query path from customer is: `orders.details.product.name`
Declaration of `orders`
```
  @OneToMany(mappedBy = "customer")
  @Where(clause = "${ta}.order_date is not null")
  List<Order> orders;
```
Declaration of `details`
```
  @Where(clause = "${ta}.id > 0")
  @OneToMany(cascade = CascadeType.ALL, mappedBy = "order")
  @OrderBy("id asc, orderQty asc, cretime desc")
  @DocEmbedded
  List<OrderDetail> details;
```
So in my mind it is correct to add the extra-where conditions here and the old code was probably wrong.

Reply:
> it is correct to add the extra-where conditions here and the old code was probably wrong.

I agree.